File: 3248-snake-in-matrix.cpp
/*
3248. Snake in Matrix

Submitted: January 17, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 34.90 MB (beats 43.05%)
*/

class Solution {
public:
    int finalPositionOfSnake(int n, vector<string>& commands) {
        int pos = 0;
        for (const string& cmd : commands) {
            pos += cmd == "UP"    ? -n : 
                   cmd == "RIGHT" ? 1 :
                   cmd == "DOWN"  ? n :
                   cmd == "LEFT"  ? -1 : 0;
        }
        return pos;
    }
};

File: 3248-snake-in-matrix.py
"""
Snake in Matrix

Submitted: January 17, 2025

Runtime: 3 ms (beats 48.96%)
Memory: 18.03 MB (beats 14.17%)
"""

class Solution:
    def finalPositionOfSnake(self, n: int, commands: List[str]) -> int:
        return sum(
            -n if cmd == "UP" else
            1  if cmd == "RIGHT" else
            n  if cmd == "DOWN" else -1
            for cmd in commands
        )